SpatialGDK/Source/SpatialGDK/Private/Interop/ClientConnectionManager.cpp

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Interop/ClientConnectionManager.h"
+
+#include "EngineClasses/SpatialNetConnection.h"
+#include "EngineClasses/SpatialPackageMapClient.h"
+#include "GameFramework/PlayerController.h"
+#include "Interop/SpatialOSDispatcherInterface.h"
+#include "Interop/SpatialReceiver.h"
+#include "SpatialView/EntityDelta.h"
+#include "SpatialView/SubView.h"
+
+DEFINE_LOG_CATEGORY(LogWorkerEntitySystem);
+
+namespace SpatialGDK
+{
+struct FSubViewDelta;
+
+ClientConnectionManager::ClientConnectionManager(const FSubView& InSubView, USpatialNetDriver* InNetDriver)
+	: SubView(&InSubView)
+	, NetDriver(InNetDriver)
+{
+}
+
+void ClientConnectionManager::Advance()
+{
+	const FSubViewDelta& SubViewDelta = SubView->GetViewDelta();
+	for (const EntityDelta& Delta : SubViewDelta.EntityDeltas)
+	{
+		switch (Delta.Type)
+		{
+		case EntityDelta::REMOVE:
+			EntityRemoved(Delta.EntityId);
+		default:
+			break;
+		}
+	}
+}
+
+void ClientConnectionManager::RegisterClientConnection(const Worker_EntityId InWorkerEntityId, USpatialNetConnection* ClientConnection)
+{
+	WorkerConnections.Add(InWorkerEntityId, ClientConnection);
+}
+
+void ClientConnectionManager::CleanUpClientConnection(USpatialNetConnection* ConnectionCleanedUp)
+{
+	if (ConnectionCleanedUp->ConnectionClientWorkerSystemEntityId != SpatialConstants::INVALID_ENTITY_ID)
```

Comment:
This can't be `INVALID_ENTITY_ID` anymore, can it?

Reply:
hmm- not sure, will see

Reply:
yeh i think you're right, removed

Reply:
turns out it totally can be invalid and removing the if here crashes the editor when a connection with an invalid entity ID makes its way in